### PR TITLE
[nxp noup] zephyr: Fix memory leak in supplicant

### PR DESCRIPTION
**Pull request by LiLongNXP**
Fix memory leak in wpa_supplicant_event_wrapper().

### CODE REVIEW

File: src/drivers/driver_zephyr.c [hunk 1]
```diff
@@ -60,8 +60,9 @@ void wpa_supplicant_event_wrapper(void *ctx,
 
 				if (!ies) {
 					wpa_printf(MSG_ERROR,
-					  "%s:%d Failed to alloc %d bytes\n",
-					  __func__, __LINE__, data->auth.ies_len);
+					  "%s:%d EVENT_AUTH %u Failed to alloc ies %d bytes\n",
+					  __func__, __LINE__, event, data->auth.ies_len);
+					os_free(msg.data);
```
Comment (krish2718) [inline src/drivers/driver_zephyr.c:65]: may be convert to a `goto` to reuse the code?

Comment (LiLongNXP) [reply src/drivers/driver_zephyr.c:65]: hi krish,
Thanks for you comment.
At beginning I have the same thought with you which perfer to use "goto" to reuse the code, but there's too much small data which define and allocate in different if blocks, then it's hard to simply do goto in current code.
Thanks.
Li Long

Comment (krish2718) [reply src/drivers/driver_zephyr.c:65]: I understand, I was thinking more like below, but yeah not really a better alternative, your approach is fine.

```
+
+out:
+       if (msg.data) {
+               if (event == EVENT_AUTH && msg.data->auth.ies)
+                       os_free(msg.data->auth.ies);
+               else if (event == EVENT_RX_MGMT && msg.data->rx_mgmt.frame)
+                       os_free(msg.data->rx_mgmt.frame);
+               else if (event == EVENT_TX_STATUS && msg.data->tx_status.data)
+                       os_free(msg.data->tx_status.data);
+               else if (event == EVENT_ASSOC && msg.data->assoc_info.addr) {
+                       os_free(msg.data->assoc_info.addr);
+                       if (msg.data->assoc_info.req_ies)
+                               os_free(msg.data->assoc_info.req_ies);
+                       if (msg.data->assoc_info.resp_ies)
+                               os_free(msg.data->assoc_info.resp_ies);
+                       if (msg.data->assoc_info.resp_frame)
+                               os_free(msg.data->assoc_info.resp_frame);
+               } else if (event == EVENT_ASSOC_REJECT && msg.data->assoc_reject.bssid) {
+                       os_free(msg.data->assoc_reject.bssid);
+                       if (msg.data->assoc_reject.resp_ies)
+                               os_free(msg.data->assoc_reject.resp_ies);
+               } else if (event == EVENT_DEAUTH && msg.data->deauth_info.addr) {
+                       os_free(msg.data->deauth_info.addr);
+                       if (msg.data->deauth_info.ie)
+                               os_free(msg.data->deauth_info.ie);
+               } else if (event == EVENT_DISASSOC && msg.data->disassoc_info.addr) {
+                       os_free(msg.data->disassoc_info.addr);
+                       if (msg.data->disassoc_info.ie)
+                               os_free(msg.data->disassoc_info.ie);
+               } else if (event == EVENT_UNPROT_DEAUTH && msg.data->unprot_deauth.sa) {
+                       os_free(msg.data->unprot_deauth.sa);
+                       os_free(msg.data->unprot_deauth.da);
+               } else if (event == EVENT_UNPROT_DISASSOC && msg.data->unprot_disassoc.sa) {
+                       os_free(msg.data->unprot_disassoc.sa);
+                       os_free(msg.data->unprot_disassoc.da);
+               }
+               os_free(msg.data);
+       }
```

**Review by krish2718 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by gannicus-gj — APPROVED** [review]

**Comment by LiLongNXP** [comment]
hi @rlubos ,
Could you please review this PR, and if ok to you please approve this PR. Thanks!

**Review by rlubos — APPROVED** [review]

**Pull request merged**